rllib/core/rl_module/rl_module.py

```diff
@@ -378,33 +401,126 @@ def get_state(self) -> Mapping[str, Any]:
     def set_state(self, state_dict: Mapping[str, Any]) -> None:
         """Sets the state dict of the module."""
 
-    def serialize(self) -> Mapping[str, Any]:
-        """Return the serialized state of the module."""
-        return {
-            "class": self.__class__,
-            "config": self.config.to_dict(),
-            "state": self.get_state(),
-        }
+    def _save_module_metadata(
+        self,
+        checkpoint_dir: Union[str, pathlib.Path],
+        module_state_path: Union[str, pathlib.Path],
+    ):
+        """Saves the metadata of the module to checkpoint_dir.
+
+        Includes:
+            - module class path
+            - module state path
+            - the module config
+            - the ray version used
+            - the ray commit hash used
+            - the date and time of the checkpoint was created
+
+        """
+        if isinstance(checkpoint_dir, str):
+            checkpoint_dir = pathlib.Path(checkpoint_dir)
+        if isinstance(module_state_path, str):
+            module_state_path = pathlib.Path(module_state_path)
+        gmt_time = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S GMT")
+        metadata = {}
+        # TODO (Avnishn): Find a way to incorporate the tune registry here.
+        metadata["module_class"] = serialize_type(self.__class__)
+        metadata["module_config"] = self.config.to_dict()
+        metadata["ray_version"] = ray.__version__
+        metadata["ray_commit_hash"] = ray.__commit__
+        metadata["checkpoint_date_time"] = gmt_time
+        metadata["module_state_path"] = str(module_state_path)
+        metadata_path = checkpoint_dir / METADATA_FILE_NAME
+        with open(metadata_path, "w") as f:
+            json.dump(metadata, f)
```

Comment:
maybe you should separate metadata computation from json file writing.
the metadata api doesn't really have anything to do with persistent file writing right?

Reply:
ok that sounds good to me. I only didn't separate them because I figured that there was no use to generating the metadata unless it was being written to a file. I'll change this now.